@xen-orchestra/lite/src/views/new-vm/NewVmView.vue

```diff
@@ -0,0 +1,968 @@
+      name_description: vdi.name_description,
+      size: bytesToGiB(vdi.virtual_size),
+      SR: vdi.SR ? getSrByOpaqueRef(vdi.SR)?.$ref : defaultSr.value,
+    })
+  })
+
+  return existingDisksArray
+}
+
+const onTemplateChange = () => {
+  const template = vmState.new_vm_template
+  if (!template) return
+
+  const { name_label, name_description, HVM_boot_params, VCPUs_at_startup, memory_dynamic_max, other_config } = template
+
+  Object.assign(vmState, {
+    name: name_label,
+    description: other_config.default_template === 'true' ? '' : name_description,
+    boot_firmware: HVM_boot_params.firmware,
+    vCPU: VCPUs_at_startup,
+    ram: memory_dynamic_max,
+    vdis: getVdis(template),
+    existingDisks: getExistingDisks(template),
+    networkInterfaces: getExistingInterface(template),
+  })
+}
+
+const vmCreationParams = computed(() => ({
+  affinityHost: vmState.affinity_host,
+  clone: isDiskTemplate.value && vmState.fast_clone,
+  existingDisks: vmState.existingDisks,
+  installMode: vmState.installMode,
+  installRepository: vmState.selectedVdi as XenApiVdi['$ref'],
+  name_label: vmState.name,
+  template: vmState.new_vm_template?.$ref,
+  vdis: vmState.vdis,
+  vifs: vmState.networkInterfaces.map(net => ({
+    network: net.interface,
+    mac: net.macAddress,
+  })),
+  cpus: vmState.vCPU,
+  vcpusMax: vmState.VCPUs_max,
+  name_description: vmState.description,
+  memory: vmState.ram,
+  autoPoweron: vmState.auto_power,
+  bootAfterCreate: vmState.boot_vm,
+  copyHostBiosStrings: vmState.boot_firmware !== 'uefi' && !templateHasBiosStrings.value && vmState.copyHostBiosStrings,
+  hvmBootFirmware: vmState.boot_firmware,
+  tags: vmState.tags,
+  cloudConfig: '',
+}))
+
+const xapi = useXenApiStore().getXapi()
+
+const createVM = async () => {
+  const templateRef = vmCreationParams.value.template
+  const newVmName = vmCreationParams.value.name_label
+  const selectedVdiRef = vmCreationParams.value.installRepository
+  const newVdis = vmCreationParams.value.vdis
+  const existingDisks = vmCreationParams.value.existingDisks
+
+  const $defer = []
+
+  if (!templateRef) {
+    console.error('Error : templateRef is undefined or invalid.')
+    return
+  }
+
+  isBusy.value = true
+
+  try {
+    const vmRefs = vmCreationParams.value.clone
+      ? await xapi.vm.clone({ [templateRef]: newVmName })
+      : await xapi.vm.copy({ [templateRef]: newVmName }, '' as XenApiSr['$ref'])
+
+    $defer.push(() => xapi.vm.delete(vmRefs))
+
+    // Removes disks from the provision XML, we will create them by ourselves.
+    await xapi.vm.removeFromOtherConfig(vmRefs, 'disks')
+
+    // Inspects the disk configuration contained within the VM's other_config,
+    // creates VDIs and VBDs and then executes any applicable post-install script.
+    await xapi.vm.provision(vmRefs)
+
+    // INSTALL SETTINGS
+    const vm = await xapi.getField<XenApiVm>('VM', vmRefs[0], 'record')
+
+    const installMethod = vmState.selectedVdi ? 'cd' : 'network'
```

Comment:
Maybe use `installMode` instead of `selectedVdi`, no?
```suggestion
    const installMethod = vmState.installMode === "ISO" ? 'cd' ? vmState.installMode === 'noConfigDrive' ? 'none' : 'network'
```
Or using a `switch/case`